tool-openssl/tool.cc

```diff
@@ -0,0 +1,120 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <string>
+#include <vector>
+#include <array>
+#include <iostream>
+
+#include <openssl/crypto.h>
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+
+#if defined(OPENSSL_WINDOWS)
+#include <fcntl.h>
+#include <io.h>
+#else
+#include <libgen.h>
+#include <signal.h>
+#endif
+
+#include "../tool/internal.h"
+#include "./internal.h"
+
+typedef bool (*tool_func_t)(const std::vector<std::string> &args);
+
+struct Tool {
+  const char *name;
+  tool_func_t func;
+};
+
+static const std::array<Tool, 2> kTools = {{
+  { "x509", X509Tool },
+  { "", nullptr },
```

Comment:
This value was only needed as an indicator to stop iterating (e.g., line 41), but it's not needed now.